distributed/client.py

```diff
@@ -1630,10 +1676,12 @@ def _handle_retried_key(self, key=None):
         if state is not None:
             state.retry()
 
-    def _handle_task_erred(self, key=None, exception=None, traceback=None):
+    def _handle_task_erred(
+        self, key=None, exception=None, traceback=None, erred_on=None, run_spec=None
+    ):
         state = self.futures.get(key)
         if state is not None:
-            state.set_error(exception, traceback)
+            state.set_error(exception, traceback, erred_on, key, run_spec)
```

Comment:
Consider:

```python
x = c.submit(lambda: 1/0, key="x")
y = c.submit(inc, x, key="y")
y.result()
```
I understand here that key="y"? However, the error happened while computing x, and x is what should appear in the error message.

Reply:
Yes, but how would you know that here? go through task dependencies? what if more than one failed?

Reply:
Ah. I see there's `exception_blame`.